x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -276,10 +286,17 @@ boolean canContinueSequentialRead(int part, long pos) {
 
         int read(byte[] b, int offset, int length) throws IOException {
             assert this.pos < maxPos : "should not try and read from a fully-read stream";
-            int read = inputStream.read(b, offset, length);
-            assert read <= length : read + " vs " + length;
-            pos += read;
-            return read;
+            int totalRead = 0;
```

Comment:
Could we extract a method to do this? The two changes to this file look very similar.

Reply:
Sure, done in e06a94b0f558ed2b85fdf1975856141a1d42210e